zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineServeTask.kt

```diff
@@ -50,18 +56,44 @@ abstract class ZiplineServeTask : DefaultTask() {
     val deploymentRegistry = services.get(DeploymentRegistry::class.java)
     val deploymentHandle = deploymentRegistry.get(deploymentId, ZiplineServerDeploymentHandle::class.java)
     if (deploymentHandle == null) {
-      val server = routes(
-        "/" bind static(Directory(inputDir.get().asFile.absolutePath), Pair("zipline", ContentType.TEXT_PLAIN))
-      ).asServer(SunHttp(port.orElse(8080).get()))
-
+      // First time this task is run, start the server
+      val ws = websockets(
+        "/ws" bind { ws: Websocket ->
+          websockets.add(ws)
+          ws.send(WsMessage(RELOAD_MESSAGE))
+          ws.onClose {
+            websockets.remove(ws)
+          }
+        }
+      )
+      val http = routes(
+        "/" bind static(
+          Directory(inputDir.get().asFile.absolutePath),
+          Pair("zipline", ContentType.TEXT_PLAIN)
+        )
+      )
+      val server = PolyHandler(http, ws).asServer(Jetty(8080))
```

Comment:
Switching from SunHttp to Jetty is a bummer. You probably lose a decent amount of start-up time here.

MAYBE we finish OkHttp 5.x someday and we can use MockWebServer here, which is fast, small, and websockets-aware.

Reply:
Yeah, I would love to use MockWebServer, then we could get good testing too. What is the dependency on OkHttp 5? Can't we add proper WS support in 4.x?